mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraACLMapper.java

```diff
@@ -66,16 +66,16 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(CassandraACLMapper.class);
 
-    public CassandraACLMapper(CassandraId cassandraId, Session session, int maxRetry) {
-        this(cassandraId, session, maxRetry, () -> {});
+    public CassandraACLMapper(CassandraId cassandraId, Session session, CassandraAsyncExecutor cassandraAsyncExecutor, int maxRetry) {
+        this(cassandraId, session, cassandraAsyncExecutor, maxRetry, () -> {});
     }
 
-    public CassandraACLMapper(CassandraId cassandraId, Session session, int maxRetry, CodeInjector codeInjector) {
+    public CassandraACLMapper(CassandraId cassandraId, Session session, CassandraAsyncExecutor cassandraAsyncExecutor, int maxRetry, CodeInjector codeInjector) {
         Preconditions.checkArgument(maxRetry > 0);
         Preconditions.checkArgument(cassandraId != null);
         this.cassandraId = cassandraId;
         this.session = session;
-        this.executor = new CassandraAsyncExecutor(session);
+        this.executor = cassandraAsyncExecutor;
```

Comment:
Yes. There's a flow with ACLMapper  as mailboxId is a field. Then we need to create it per request. I did not wanted some weird threadpool issues.